Instrument shield add-on install funnel

rayborn opened the pull request:
First PR for amo, so please give a through review.  

Our installation numbers are pretty low in Shield Study 1, but we didn't have the whole funnel instrumented.  This iteration instruments every stage of the funnel to see where we're losing interest.

RE: https://bugzilla.mozilla.org/show_bug.cgi?id=1264008

muffinresearch commented on src/olympia/pages/templates/pages/shield_study_2.html:
```diff
@@ -26,7 +29,7 @@
         <h1 data-hook="title">Help Make Firefox Faster</h1>
         <div class="experiment-controls">
-          <a class="button primary large" href="https://addons.mozilla.org/firefox/downloads/latest/x-screen-performance-study-1/">Start Performance Study</a>
+          <span id="opt-in" class="button primary large">Start Performance Study</span>
```
Is there a reason to not have a real link here? It might be nice to link through to something even if JS was disabled. Also spans aren't focusable for people that use a keyboard to navigate.

rayborn replied:
This was an attempt to treat this like a button with no context menu.  Eventually we may be able to install from the Shield addon itself, so I wanted this to seem more "native" in feel.  Although now that I actually research it looks like the "correct" was to do this is setting oncontextmenu.  That said, I'm happy to go back to a hyperlink; it was a minor detail to me.

muffinresearch commented on static/js/shield_study/main.js:
```diff
@@ -0,0 +1,15 @@
+    event.target.innerHTML = "Thank you! Opted In";
+    event.target.setAttribute("onclick","");
+    if (action) {
+      window.location.href = "https://addons.mozilla.org/firefox/downloads/latest/x-screen-performance-study-1/";
```
What's the rationale behind randomly sending people to this page some of the time? Should something else happen when action is false?

rayborn replied:
We wanted to treat this as an A/B test since the Shield addon will be able to install secondary addons without the scary addon installation prompt.  We wanted to see how this step of the funnel is affected by no message.

However, I'm currently trying to convince my coworkers that we don't need to structure this as an A/B test since the no-installation branch is the same as the installation branch less the last step.  Unless I'm missing something I'd like to delete this.  Will update shortly

muffinresearch commented:
I see the bug mentions optimizely. Are you planning to add that? 

We had issues in the past with optimizely not being compatible with CSP so if you are planning to add it we'd need to check that out carefully - I know they planned to make changes but I'm not sure that's happened yet.

rayborn commented:
We decided to go with GA since 1) this is a very simple task, 2) I'm more familiar with GA, and 3) you already had GA in place.

To give additional context for rational:  Our last study had very poor opt-in so we wanted to do a run with a very small sample with full instrumentation.  Our hypothesis is that the add-on installation chrome warning is scaring users away who may feel like this is phishing despite the amo url.  We could bypass this issue using the Shield system add-on to manage these tests in the future and want to evaluate how that change would effect our funnel.

muffinresearch commented on src/olympia/pages/templates/pages/shield_study_2.html:
```diff
@@ -26,7 +29,7 @@
         <h1 data-hook="title">Help Make Firefox Faster</h1>
         <div class="experiment-controls">
-          <a class="button primary large" href="https://addons.mozilla.org/firefox/downloads/latest/x-screen-performance-study-1/">Start Performance Study</a>
+          <a id="opt-in" class="button primary large" oncontextmenu="return false" href="https://addons.mozilla.org/firefox/downloads/latest/x-screen-performance-study-1/">Start Performance Study</a>
```
The inline event is causing a CSP violation.

rayborn replied:
I'll move it to the js.  Thanks

muffinresearch commented on src/olympia/pages/urls.py:
```diff
@@ -48,7 +48,7 @@
         name='pages.sunbird'),
     url('^sunbird/', lambda r: perma_redirect(reverse('pages.sunbird'))),
 
-    url('^shield-study-1/',
```
I presume nothing will continue to link here?

rayborn replied:
Good catch. All entry points to that study have been discontinued.  Should I remove the corresponding HTML page in this PR or leave the path in urls.py for a later cleanup?

muffinresearch replied:
Looks like you're renaming the template in this PR. I suppose the only thing is if anyone was linking the previous URL. If 404s aren't a problem to you then removing this is OK assuming there's not a bunch of backlinks pointing at it.

rayborn replied:
All previous study links are wrapped up and I can't imagine anyone of the users we surveyed bookmarked it.  Should be fine from my perspective.

rayborn commented:
@muffinresearch @andymckay 
I think that I've addressed all the comments. 

Also, I really appreciate the good ReadTheDocs documentation.  IMO it's pretty rare to be able to stand up a local test server without running into some confusing directions.

Thanks and let me know if you have additional questions/comments.

EnTeQuAk commented:
I restarted travis build, the failure was not related to the PR

rayborn commented:
Oh good, I was a bit confused given the minor changes in the most recent commit.

Thanks

muffinresearch commented:
@rayborn looks good.  I'll go ahead and merge this and it should be ready to test on https://addons-dev.allizom.org in due course. If you need to expedite any changes just ping me on irc. As I mentioned we tag at 4.00pm UTC today (tuesday) and so this will go out on Thursday at a similar time.
